### PR TITLE
docs(blueprints): improve description of blueprint

### PR DESCRIPTION
**Pull request by ingeridhellen**
## Issues related to this change

#35

### CODE REVIEW

File: docs/docs/concepts/blueprints.md [hunk 1]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
-* It has the two attributes `name` and `description`.
+- Line 2 contains the name of the blueprint, which is used when referring to this blueprint.
+- Line 3 contains the type of the blueprint. The word `type` carries some extra meaning in the Development Framework since it tells us which blueprint a blueprint or entity is implementing. In this case, it tells us that this JSON object is of type [dmss://system/SIMOS/Blueprint][Blueprint], and will therefore have attributes such as `name`, `type`, `description` and `attributes`. If you open the URL, you can see that it links to a blueprint with name Blueprint and type `dmss://system/SIMOS/Blueprint`. That means that the Blueprint blueprint actually implements itself!
+- Line 4 contains the description of the blueprint, which tells us that this blueprint can be used to describe a car.
+- Line 5 to 21 list the attributes of the blueprint - `type`, `carModel` and `registrationNumber`. All entities implementing this blueprint must have these three attributes, and only them. An attribute definition is always an implementation of the [dmss://system/SIMOS/BlueprintAttribute][BlueprintAttribute], hence the type.
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md:51]: Is this easier to understand?
```suggestion
- Line 5 to 21 list the attributes of the blueprint - `type`, `carModel` and `registrationNumber`. All entities of type `Car` must have these three attributes, and only them. An attribute definition is always an implementation of the [dmss://system/SIMOS/BlueprintAttribute][BlueprintAttribute], hence the type.
```

File: docs/docs/concepts/blueprints.md [hunk 2]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
-   "name": "Volvo",
-   "description": "This is an typically car type used in Norway"
+  "type": "./Car",
+  "carModel": "Skoda Octavia",
+  "registrationNumber": "AB12345"
 }
 ```
 
-By looking at the `type` keyword, we can see that it's of type `Car`.
+Notice how the `type` attribute refers to the name attribute of the blueprint we wrote above? For more info about how to reference blueprints, see [References to other blueprints](#references).
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md:69]:
```suggestion
Notice how the `type` attribute refers to the name attribute of the blueprint we wrote above? For more info about how to reference blueprints, see [References to other blueprints](#references).

We can see that all attributes in the entity ("type", "carModel", and "registrationNumber") are defined in the Car blueprint's attributes list. 
```

File: docs/docs/concepts/blueprints.md [hunk 3]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
-| extends       | string                            | A list of references to other blueprints. The blueprint will inherit attributes from the referenced blueprints. |
-| attributes    | `system/SIMOS/BlueprintAttribute` | List of attributes.                                                                                             | 
+| Attribute   | Type                              | Description                                                                                                     |
+| ----------- | --------------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| type        | string                            | A reference to the Blueprint type.                                                                              |
+| name        | string                            | The name of the blueprint                                                                                       |
+| description | string                            | More information about the model                                                                                |
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md]:
```suggestion
| description | string                            | More information about the blueprint                                                                                |
```

File: docs/docs/concepts/blueprints.md [hunk 4]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
+| extends     | string                            | A list of references to other blueprints. The blueprint will inherit attributes from the referenced blueprints. |
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md]:
```suggestion
| extends     | string                            | A list of references to other blueprints. The blueprint will inherit all attributes from the blueprint(s) it extends from. |
```

File: docs/docs/concepts/blueprints.md [hunk 5]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
+| type          | string   | A reference to the BlueprintAttribute type.                                                                                                                                                                                                                          |
+| description   | string   | Extra information                                                                                                                                                                                                                                                    |
+| attributeType | string   | The attribute type. Can be a reference to another blueprint, or a primitive type (string, integer, number, boolean).                                                                                                                                                 |
+| dimensions    | string   | Degree of dimensions, formatted as a comma separated list of integers or `*`. `*`(wildcard) has the special meaning of "unfixed size". Ex: `*` (1D of infinite length), `2,2` (2D of size 2x2), `3,*` (2D, where the inner array have infinite length)               |
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md]:
```suggestion
| dimensions    | string   | Degree of dimensions, formatted as a comma separated list of integers or `*`. The `*` (wildcard) has the special meaning of "unknown length". Ex: `*` (1 dimensional array of unknown length), `2,2` (2 dimensional array of size 2x2), `3,*` (2 dimensional array, where the inner array have unknown length and the outer array have length 3)               |
```

File: docs/docs/concepts/blueprints.md [hunk 6]
```diff
@@ -4,145 +4,149 @@ sidebar_position: 1
...
-## References 
+## References to other blueprints {#references}
 
 It is possible to reference other blueprint from the `type`, `extends`, `attributeType`, and `_blueprintPath_` attributes.
 
-### Syntax
+There are several ways to describe a reference to other blueprints, all of which are understood and resolved during import.
 
-There are several ways to describe a reference to other blueprints. 
+Note that the reference is created using the name attribute of the blueprint you want to refer to, not its filename. Ie "Car", not "Car.blueprint.json".
```
Comment (KristianKjerstad) [inline docs/docs/concepts/blueprints.md]:
```suggestion
Note that the reference is created using the name attribute of the blueprint you want to refer to, not its filename. I.e. "Car", not "Car.blueprint.json".
```

**Review by KristianKjerstad — APPROVED** [review]
nice, approved with some suggestions

File: docs/docs/concepts/blueprints.md (hunk 1 above)
Comment (ingeridhellen) [reply docs/docs/concepts/blueprints.md:51]: Not sure. The way we refer to a blueprint is a bit confusing (ie why `Car` and not `Car.blueprint.json`). I added a comment about it later on, but I'm not sure explicitly assuming the reader understands it at this point is wise.

File: docs/docs/concepts/blueprints.md (hunk 2 above)
Comment (ingeridhellen) [reply docs/docs/concepts/blueprints.md:69]: Rewritten slightly and added

**Pull request merged**